### DISCUSSION TITLE
Define configuration externally

### DISCUSSION THREAD

**Discussion by britanik** [oidc-wp/openid-connect-generic#314] [Q&A] [answered]
Hello, how can I define configuration strings like:

- Client ID
- Client Secret Key
- Endpoints

externally, in php constants or server vars. Can you point me in the right direction? ☺️
Thanks!

**Comment by daggerhart** [comment, marked as answer]
Hi britanik,

You can define them with the constants noted here-- https://github.com/oidc-wp/openid-connect-generic/blob/dev/includes/openid-connect-generic-option-settings.php#L88

* OIDC_CLIENT_ID
* OIDC_CLIENT_SECRET
* OIDC_ENDPOINT_LOGIN_URL
* OIDC_ENDPOINT_USERINFO_URL
* OIDC_ENDPOINT_TOKEN_URL
* OIDC_ENDPOINT_LOGOUT_URL
